enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,549 @@
+memory requests the quality-of-service class of the pod would be
+changed automatically. Any labeled pod that is already BestEffort
+would be annotated using `0` as the value so that CRI-O will have an
+indicator to configure the CPU shares as BestEffort.
+
+#### Kubelet Changes
+
+Kubelet will be changed to look for a configuration file,
+`/etc/kubernetes/management-pinning`, to enable the management
+workload partitioning feature. The file should contain the cpuset
+specifier for the CPUs making up the management CPU pool.
+
+Kubelet will be changed so that when the feature is enabled, [when a
+static pod definition is
+read](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/config/file_linux.go)
+it is mutated in a way similar to the API server admission hook will
+mutate regular pods.
+
+Kubelet will be changed so that when the feature is enabled, it
+advertises a new extended resource of "management cores", representing
+all of the CPU capacity of the host (not just the management CPU
+pool).
+
+#### Installer Changes
+
+The installer will be changed to accept the new `managementCPUIDs`
+configuration input value.
+
+The installer will be changed to generate an extra machine config
+manifest to configure CRI-O so that containers from pods with the
+`io.openshift.management.cores` annotation are run on the
+`mgmt_ctr_cpuset`.
+
+The installer will be changed to create a machine config manifest to
+write the `/etc/kubernetes/management-pinning` configuration file for
+kubelet. The file will have SELinux settings configured so that it is
+only readable by the kubelet.
+
+The installer will be changed to update the
+`status.managementWorkloadPartitioning` field of the Infrastructure
+config resource to `enabled` so the admission hook will know when to
+mutate pods.
+
+### Risks and Mitigations
+
+The first version of this feature must be enabled when a cluster is
+deployed to work correctly. However, some of the enabling
+configurations could be modified by an admin on day 2. We need to
+document clearly that this is not supported. In particular, changing
+the configuration and rebooting may cause normal pods to be isolated
+but will not configure kubelet to also partition static pods.
+
+It is possible to build a cluster with the feature enabled and then
+deploy an operator or other workload that should take advantage of the
+feature without first configuring the namespace properly.  We need to
+document that the configuration only applies to pods created after it
+is set, so if an operator is installed before the namespace setting is
+changed the operator or its operands may need to be re-installed or
+restarted.
+
+The schedule for delivering this feature is very aggressive. We have
+tried to minimize the number of places that need complex changes to
+make it more likely that we can meet the deadline.
+
+Many implementation details proposed here require us to carry patches
```

Comment:
i have desired the ability to let the kubelet manage and launch pods that should have run in the reservation resource set.  for static pods in particular, this capability is very sensible to pursue upstream imo.  the pattern of cpu->management conversion if not applicable upstream is something we may want to publish for others in community to adopt similar to intel and nokia and others who have all explored cpu management policies outside of core k8s.  this has a benefit of being entirely complementary to the in-tree cpu manager.